### PR TITLE
build(aio): upgrade to 5.2.0-rc.0

### PR DESCRIPTION
**Pull request by IgorMinar**

### CODE REVIEW

**Comment by IgorMinar** [comment]
uh oh. it looks like there is a regression in core, post likely in how we handle classes. @mhevery @matsko do you know of any change related to that?

The failing test:
https://github.com/IgorMinar/angular/blob/e6ec8e878b6d088857d8b58e1ab43f6066d43ca6/aio/src/app/embedded/code/code-tabs.component.spec.ts#L77-L85

The failure:

```
Chrome 64.0.3282 (Mac OS X 10.12.6) CodeTabsComponent (tab labels) should add the `class` to the label element FAILED
        Expected $[0] = 'foo-class ng-star-inserted' to equal 'foo-claxss'.
        Expected $[1] = 'bar-class ng-star-inserted' to equal 'bar-class'.
            at UserContext.<anonymous> Users/iminar/Dev/angular/aio/src/app/embedded/code/code-tabs.component.spec.ts:84:23)
            at ZoneDelegate.webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invoke Users/iminar/Dev/angular/aio/node_modules/zone.js/dist/zone.js:388:1)
            at ProxyZoneSpec.webpackJsonp.../../../../zone.js/dist/proxy.js.ProxyZoneSpec.onInvoke Users/iminar/Dev/angular/aio/node_modules/zone.js/dist/proxy.js:79:1)
            at ZoneDelegate.webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invoke Users/iminar/Dev/angular/aio/node_modules/zone.js/dist/zone.js:387:1)
```

**Comment by petebacondarwin** [comment]
@IgorMinar it looks like the animations library is adding extra class names now: https://github.com/angular/angular/commit/c2b3792a3b5fa5215fe2ef7e0ac6511086ffe4c1

I don't think this is a regression, we just need to relax the expectation, no?

File: aio/scripts/payload.sh [hunk 1]
```diff
@@ -8,5 +8,5 @@ readonly parentDir=$(dirname $thisDir)
 # Track payload size functions
 source ../scripts/ci/payload-size.sh
 
-trackPayloadSize "aio" "dist/*.bundle.js" true true "${thisDir}/_payload-limits.json"
+trackPayloadSize "aio" "dist/*.js" true true "${thisDir}/_payload-limits.json"
```
Comment (gkalpak) [inline aio/scripts/payload.sh]: Should we define limits for non-bundle JS files?

**Review by gkalpak — APPROVED** [review]
LGTM (as soon as Travis is green)

**Comment by IgorMinar** [comment]
@gkalpak any idea why I'm getting this error: 
```
Request to overwrite existing public directory: /var/www/aio-builds/21351/40f4789
HTTP_CODE: 409
Running PWA audit for 'https://pr21351-40f4789.ngbuilds.io'...
{ Error: connect ECONNREFUSED 127.0.0.1:41437
    at Object._errnoException (util.js:1024:11)
    at _exceptionWithHostPort (util.js:1046:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1182:14)
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 41437 }
error Command failed with exit code 1.
```
https://travis-ci.org/angular/angular/jobs/326687199#L1212

**Comment by IgorMinar** [comment]
hmm.. it seems that other of my PRs are failing with a similar error:

```
Running PWA audit for 'https://pr21350-f0762aa.ngbuilds.io'...
{ Error: connect ECONNREFUSED 127.0.0.1:35845
```
from https://travis-ci.org/angular/angular/jobs/326686272#L1216 and https://github.com/angular/angular/pull/21350

**Comment by gkalpak** [comment]
I suspect it is because of https://blog.travis-ci.com/2018-01-08-travis-response-meltdown-spectre.
They are [in the process][1] of routing jobs back to the container based infrastructure (which should fix the issue, I hope).

Related Travis issue: https://github.com/travis-ci/travis-ci/issues/9024

[1]: https://www.traviscistatus.com/incidents/7k5kpkcldbnk

File: aio/scripts/payload.sh (hunk 1 above)
Comment (IgorMinar) [reply aio/scripts/payload.sh]: this didn't do what I thought it would do. I'll take the commit out.

Comment (IgorMinar) [reply aio/scripts/payload.sh]: I wanted to track the size without imposing any limit.

**Comment by IgorMinar** [comment]
@gkalpak and @petebacondarwin I throw in a commit that turns on `namedChunks`. I hope you don't mind.

**Comment by gkalpak** [comment]
Interestingly, the preview is broken for me 😕
I get:

```
VM84:34 ERROR TypeError: r.a.timer is not a function
    at t.initWorker (main.7e1fc2e18ff98132f321.bundle.js:formatted:131)
    at t.ngOnInit (main.7e1fc2e18ff98132f321.bundle.js:formatted:13641)
    at main.7e1fc2e18ff98132f321.bundle.js:formatted:7657
    at oi (main.7e1fc2e18ff98132f321.bundle.js:formatted:7749)
    at Ii (main.7e1fc2e18ff98132f321.bundle.js:formatted:8082)
    at Object.updateDirectives (main.7e1fc2e18ff98132f321.bundle.js:formatted:15815)
    at Object.updateDirectives (main.7e1fc2e18ff98132f321.bundle.js:formatted:8553)
    at Object.ii [as checkAndUpdateView] (main.7e1fc2e18ff98132f321.bundle.js:formatted:7551)
    at t.detectChanges (main.7e1fc2e18ff98132f321.bundle.js:formatted:6595)
    at main.7e1fc2e18ff98132f321.bundle.js:formatted:4660
```

I see that the tests passed on the CI (even the Lighthouse test, which runs against the deployed version), but something is wrong (somewhere).

**Comment by gkalpak** [comment]
Actually, even with this error, PWA tests pass (because we serve an empty (apparently very performant) page :)
And our unit/e2e tests don't catch this (possibly because it only happens in `--prod` mode (e.g. cause by the build-optimizer - TBD).

Looking...

**Comment by gkalpak** [comment]
I didn't look into what exactly was causing the error (it seems like the `Observable.timer()` method was not included in the bundle), but [this commit][1] fixed it locally.
I also pushed a commit to upgrade `compiler-cli` to 5.2.0. Was it intentionally left to the old version, @IgorMinar? 

[1]: https://github.com/angular/angular/pull/21351/commits/28bf191142dbc0471b83098f8e50deff871524b4

**Comment by IgorMinar** [comment]
@gkalpak omg, you rock! thanks for catching this. it's odd that build-optimizer removes those imports with side-effects.

@filipesilva this sounds like a recent regression - we can't mark `import from 'rxjs/add/operator/foo'`  as  pure. Do you know why this is happening?

@gkalpak I'll squash your changes into mine.

**Comment by filipesilva** [comment]
@IgorMinar I'm sorry, I don't follow. Can you point me to where the regression is being shown?

**Comment by IgorMinar** [comment]
@filipesilva please take a look at

- https://github.com/angular/angular/pull/21351#issuecomment-356573314
- https://pr21351-e62d89d.ngbuilds.io/
- https://github.com/angular/angular/pull/21351/commits/ea976255f30b69328127eb19866d64788e147856

**Comment by filipesilva** [comment]
I understand the context now. I do not know what is problem is, no... Can someone setup a repro and issue on https://github.com/angular/devkit?

**Comment by filipesilva** [comment]
I tried a naive repro on a new app, using Angular 5.1.3 and CLI 1.6.3, with a `ng serve --prod`:
```
// main.ts
import { Observable } from 'rxjs/Observable';
import 'rxjs/add/observable/timer';
console.log(Observable.timer);
```

The console.log printed a function.

Then I updated to Angular 5.2.0 and the result was the same.

Not sure why it's different on AIO.

**Pull request closed without merging**